content/systems/filecoin_blockchain/storage_power_consensus/_index.md

```diff
@@ -93,147 +88,42 @@ lowest epoch number block will contain the requested beacon entry. Similarly, if
 there has been null rounds where the beacon should have been inserted, we need
 to iterate on the chain to find where the entry is inserted.
 
-```go
-func GetRandomnessFromBeacon(e ChainEpoch, head ChainEpoch) (DrandEntry,error) {
-  // get the drand round associated with the timestamp of this epoch.
-  drandRound := MaxBeaconRoundForEpoch(e)
-  // get the minimum drand timestamp associated with the drand round
-  drandTs := drandGenesisTime + (drandPeriod-1) * drandRound 
-  // get the minimum filecoin epoch associated with this timestamp
-  minEpoch := (drandTs - filGenesisTime) / filEpochDuration 
-  for minEpoch < head {
-     // if this is not a null block, then it must have the entry we want
-    if !chain.IsNullBlock(minEpoch) 
-         // the requested drand entry must be in the list of drand entries
-         // included in this block. If it is not the case, 
-         // it means the block is invalid - but this condition is caught by the 
-         // block validation logic.
-         returns getDrandEntryFromBlockHeader(chain.Block(minEpoch))
-    // otherwise, we need to continue progressing on the chain, i.e. maybe no
-    // miner were elected or filecoin / drand outage
-    minEpoch++
-  }
-}
-
-func getDrandEntryFromBlockHeader(block,round) (DrandEntry,error) {
-    for _,dr := range block.DrandEntries {
-        if dr.Round == round {
-            return dr
-        }
-    }
-    return errors.New("drand entry not found in block")
-}
-```
+
+{{<embed src="https://github.com/filecoin-project/lotus/blob/master/chain/vm/runtime.go" lang="go" symbol="GetRandomnessFromBeacon">}}
```

Comment:
This is currently invalid in Lotus as described in here https://github.com/filecoin-project/research-private/issues/364